src/field/MagField.hh

```diff
@@ -0,0 +1,45 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file MagField.cuh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The MagField evaluates the magnetic field value at a given position.
+ * Support: 1) a constant magnetic field
+ *          2) a user defined field (e.g.: field map or parameterized field)
+ *             along with MagFieldPointers and MagFieldStore: XXX TODO
+ */
+class MagField
+{
+  public:
+    //@{
+    //! Type aliases
+    using field_value = Real3;
+    //@}
+
+  public:
+    // Construct from a uniform field
+    CELER_FUNCTION MagField(const field_value& value);
+
+    // return a uniform magnetic field value
+    CELER_FUNCTION field_value operator()();
```

Comment:
Should this eventually be a function of position and/or volume? We could put those in as a placeholder showing the eventual intent, and just not use them for the uniform case.

Reply:
Yes, both constructor and the field retrieving method should be extended in the next round (underway).  Also this class will be served a base for the user magnetic field class.  I just added a dummy method with a position argument as suggested for this MR.